Remove Warning and add notice for Navigation

AKSHAT2802 opened the pull request:
Fixes https://github.com/WordPress/gutenberg/issues/59785

<img width="276" alt="Screenshot 2024-07-25 at 1 05 25 PM" src="https://github.com/user-attachments/assets/3a401ba8-713c-4008-a15f-4bcac404ad4e">
<img width="674" alt="Screenshot 2024-07-25 at 1 05 53 PM" src="https://github.com/user-attachments/assets/2b6df621-ad03-404f-bd7d-d696b0d87895">

AKSHAT2802 committed:
Remove Warning and add notice for Navigation

Mamaduka commented on packages/block-library/src/navigation/edit/deleted-navigation-warning.js:
```diff
@@ -1,24 +1,33 @@
+		noticeVisible && (
+			<Notice
+				status="warning"
+				onRemove={ () => setNoticeVisible( false ) }
+				isDismissible
```
I think this shouldn't be dismissible.

afercia replied:
I'd agree it shouldn't.

afercia reviewed (changes requested):
@AKSHAT2802 thanks for working on this and sorry for the late review. In the issue https://github.com/WordPress/gutenberg/issues/59785 I proposed to change only the Warning in the Inspector panel, while this PR changes also the one in the editor canvas. Screenshot:

![Screenshot 2024-09-10 at 08 38 38](https://github.com/user-attachments/assets/80e2d8df-a5b4-4aca-85ea-984a8753b39d)

Also, I'd agree with @Mamaduka that the Notice should not be dismissible.

Could you please have a look and address both points, when you have a chance?

AKSHAT2802 committed:
Merge branch 'trunk' of https://github.com/AKSHAT2802/gutenberg into fix/replace-deleted-navigation-menu-warning-with-notice

AKSHAT2802 committed:
Address changes

AKSHAT2802 commented:
Hey @afercia, Addressed both points and updated my PR. 
As both the editor and inspector panel deleted the navigation warning from one file, I added a prop named `isNotice` and passed it true while triggering for the inspector panel. 

This will only change the warning to non-dismissable notice for the inspector panel.
I have verified that it only occurs for editor and inspector control.
PFA: Screenshot attached 
<img width="1232" alt="Screenshot 2024-09-11 at 1 35 25 AM" src="https://github.com/user-attachments/assets/6c59753a-6b71-4a94-976e-2eabe7fe6575">

afercia commented on packages/block-library/src/navigation/edit/deleted-navigation-warning.js:
```diff
@@ -2,12 +2,29 @@
+function DeletedNavigationWarning( { onCreateNew, isNotice = false } ) {
+	return isNotice ? (
+		<Notice status="warning" isDismissible={ false }>
+			{ createInterpolateElement(
```
To avoid code duplication, the whole `createInterpolateElement()` should be assigned to a `const` that is then used for both the Notice and the Warning.

afercia reviewed (changes requested):
@AKSHAT2802 thanks for the update and the screenshot. Looks good to me, I'd only reduce doce duplication. See inline comment.

AKSHAT2802 committed:
Remove redundant code and add const

AKSHAT2802 commented:
Thanks for the review @afercia & @Mamaduka 
I have added a commit to reduce duplication.

afercia reviewed (approved):
@AKSHAT2802 thank you. LGTM to me. Can be merged when all tests pass.
